sdks/python/apache_beam/io/textio_test.py

```diff
@@ -433,6 +433,80 @@ def test_read_gzip_empty_file(self):
     assert_that(pcoll, equal_to([]))
     pipeline.run()
 
+  def _remove_lines(self, lines, sublist_lengths, num_to_remove):
+    """Utility function to remove num_to_remove lines from each sublist.
+
+    Args:
+      lines: list of items.
+      sublist_lengths: list of integers representing length of sublist
+        corresponding to each source file.
+      num_to_remove: number of lines to remove from each sublist.
+    Returns:
+      remaining lines.
+    """
+    curr = 0
+    result = []
+    for offset in sublist_lengths:
+      end = curr + offset
+      start = min(curr + num_to_remove, end)
+      result += lines[start:end]
+      curr += offset
+    return result
+
+  def _read_skip_header_lines(self, file_or_pattern, skip_header_lines):
+    """Simple wrapper function for instantiating TextSource."""
+    source = TextSource(
+        file_or_pattern,
+        0,
+        CompressionTypes.UNCOMPRESSED,
+        True,
+        coders.StrUtf8Coder(),
+        skip_header_lines=skip_header_lines)
+
+    range_tracker = source.get_range_tracker(None, None)
+    return [record for record in source.read(range_tracker)]
+
+  def test_read_skip_header_single(self):
+    file_name, expected_data = write_data(TextSourceTest.DEFAULT_NUM_RECORDS)
+    assert len(expected_data) == TextSourceTest.DEFAULT_NUM_RECORDS
+    skip_header_lines = 1
```

Comment:
Also add following tests.

(1) Read a file after performing a non-trivial initial splitting with skip_header_lines > 0
(2) Run a pipeline that uses textio.ReadFromText where skip_header_lines > 0
(3) Read a compresses text file with skip_header_lines > 0